pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package cniserver
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/plugins/pkg/ip"
+	"github.com/containernetworking/plugins/pkg/ipam"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/j-keck/arping"
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+	"github.com/vmware-tanzu/antrea/pkg/agent/util/ethtool"
+	cnipb "github.com/vmware-tanzu/antrea/pkg/apis/cni/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig"
+)
+
+type ifConfigurator struct {
+	ovsDatapathType string
+}
+
+func newIfaceConfigurator(ovsDatapathType string) interfaceConfigurator {
+	return &ifConfigurator{ovsDatapathType: ovsDatapathType}
+}
+
+// initialize returns nil immediately on Linux Node.
+func (ic *ifConfigurator) initialize() error {
```

Comment:
Humm - is it good to merge this one to the new func?

Reply:
`new` is to construct an object, and initialize() is used to load existing resources which might hit errors on Windows. And I think it would be better to check errors in initialization phase but not in `new`.

Reply:
I feel it is ok to call new() in podConfigurator.initialize(), or cniServer.initialize() calls  podConfigurator.new() which calls ifConfigurator.new(). I think it is not fixed pattern that new func can not return errors.
But not strong preference on this. I am fine with a separate initialize() too. Just try to see if we can reduce some passthrough funcs/calls.

Reply:
Agree with @jianjuns  that no fixed pattern that newXX can only construct struct and not return error.
It's quite common that newXX does initialization work including external call which might fail. (Search `func New.*error` in go src can find lots of examples)  I think the most common usage of a separate Initialization is when you have to delay some initialization work or wait for some dependency which is not ready when constructing the struct. 
I'm not against to have Initialization too, just to remind we don't need to separate them when it makes things more complex.

Reply:
Think we could merge podConfigurator.new() and initialize() too, and let cniServer.initialize() call new(). But again up to you to decide.